### DISCUSSION TITLE
How to create translations for page furniture

### DISCUSSION THREAD

**Discussion by drcongo** [openownership/openownership.org-wagtail#270] [How to...]
Some text isn't translatable in the CMS, but can still be translated for the different locales by editing the site-wide `.po` files. These can all be found inside [/app/locale/](https://github.com/hactar-is/openownership.org/tree/staging/app/locale). The system will automatically compile these files into the machine readable `.mo` versions upon deploy to either the staging server or the production server.

Because we have continuous integration set up, this system means you're not reliant on Hactar to add a new translation for you, you can edit the file directly in GitHub and commit it to the `staging` branch to trigger CI to the staging server. 

Note: please don't commit to the `main` branch - instead commit to `staging` and if you need it on production as soon as possible you can open a pull request.

Side note for Hactar: Generate the initial .po files with `manpy makemessages --all -e html,jinja,txt,py`

**Comment by StephenAbbott** [comment]
@drcongo Thank you for jotting down all these notes. Am I right in thinking that this approach applies to main site furniture but doesn't allow us to provide translated versions of objects/sub links from Navbar blocks listed/created in https://www.openownership.org/admin/settings/settings/navigationsettings/1/ ?

Wondering if there is a way for us to translate the left-hand navigation menu appearing on pages like https://www.openownership.org/fr/publications (on the left on desktop, hidden behind dropdown menu on mobile)
